features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx

```diff
@@ -0,0 +1,191 @@
+import TextField from '@material-ui/core/TextField';
+import PropTypes from 'prop-types';
+import Checkbox from '@material-ui/core/Checkbox';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import { withStyles } from '@material-ui/core/styles';
+import { Grid, Paper, Typography, Divider } from '@material-ui/core';
+import { FormattedMessage } from 'react-intl';
+import { Progress } from 'AppComponents/Shared';
+import Alert from 'AppComponents/Shared/Alert';
+import API from 'AppData/api';
+import BusinessPlans from './BusinessPlans';
+
+const styles = theme => ({
+    root: {
+        flexGrow: 1,
+    },
+    container: {
+        display: 'flex',
+        flexWrap: 'wrap',
+    },
+    margin: {
+        margin: theme.spacing.unit,
+    },
+    paper: {
+        padding: theme.spacing.unit * 2,
+        textAlign: 'left',
+        color: theme.palette.text.secondary,
+        paddingBottom: '10px',
+    },
+    grid: {
+        paddingLeft: '10px',
+        paddingRight: '10px',
+        paddingBottom: '10px',
+        minWidth: '50%',
+    },
+    button: {
+        margin: theme.spacing.unit,
+    },
+});
+
+class Monetization extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            properties: [],
+            monStatus: null,
+            property: {},
+        };
+        this.handleChange = this.handleChange.bind(this);
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    componentDidMount() {
+        const { api } = this.props;
+        api.getSettings().then((settings) => {
+            if (settings.MonetizationProperties != null) {
+                this.setState({ properties: settings.MonetizationProperties });
+            }
+        });
+        api.getMonetization(this.props.api.id).then((status) => {
+            this.setState({ monStatus: status.enabled });
+        });
+    }
+
+    /**
+     * Handles the submit action for configuring monetization
+     */
+    handleSubmit() {
+        const { api } = this.props;
+        const properties = this.state.property;
+        const enabled = this.state.monStatus;
+        const body = JSON.stringify({
+            enabled,
+            properties,
+        });
+        const promisedScopeAdd = api.configureMonetizationToApi(this.props.api.id, body);
+        promisedScopeAdd.then((response) => {
+            if (response.status !== 200) {
+                Alert.info('Something went wrong while configuring monetization');
+                return;
+            }
+            Alert.info('Monetization Configured Successfully');
+        }).catch((error) => {
+            console.error(error);
+            if (error.response) {
+                Alert.error(error.response.body.message);
+            } else {
+                Alert.error('Something went wrong while configuring moneization');
```

Comment:
Shall we add i18n to all the alert messages?

Reply:
Here is our developer guide for `i18n`

https://github.com/wso2/carbon-apimgt/wiki/How-internationalization-(i18n)-works-in-API-Manager-React-Apps

Reply:
added in #b29eb6f